src/@next/Banner/Banner.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { ButtonGroup } from '../ButtonGroup';
+import { Icon } from '../Icon';
+import { IconNames } from '../Icon/icons/icons';
+import { Typography } from '../Typography';
+import {
+  StyledBanner,
+  StyledBannerContentContainer,
+  StyledBannerTitleContainer,
+  StyledCloseIconWrapper,
+} from './BannerStyle';
+
+export type BannerProps = {
+  title?: string;
+  /** If set will take precedent of status */
+  iconName?: IconNames;
+  status?: 'success' | 'info' | 'warning' | 'critical';
+  children?: React.ReactNode;
+  action?: React.ReactNode;
+  secondaryAction?: React.ReactNode;
+  dismissable?: boolean;
+  onDismiss?: () => void;
+};
+
+const iconNameStatusMap = {
```

Comment:
maybe we can type this so we don't need to cast the type of `iconNameValue`. we can use `Partial<IconNames>` maybe?

Reply:
Yeah good call!